### ISSUE TITLE
Climo files do not honor start_year & end_year if starting from timeseries files

### ISSUE THREAD

**Issue by brianpm** [NCAR/ADF#107]
### ADF run type

Model vs. Model

### What happened?

Reported by @judithberner. When starting from pre-generated time series files, ADF tries to make a climo file from the full time series no matter whether start_year and end_year are provided. 

### ADF Hash you are using

1c7d1ce

### What machine were you running the ADF on?

CGD machine

### What python environment were you using?

Personl Conda env

### Extra info

I think @judithberner was running from the standard ADF (she can correct me if that isn't the case). 

Example yaml file that produces issue: 
`/glade/work/berner/ADF/config_cam_timebounds_bug.yaml`

I think I've got a simple fix in `averaging_example.py` that should correct the issue (around line 138):

```python
            # Need to check whether averaging interval is supplied
            if ('start_year' in calc_climos):
                sy = calc_climos['start_year']
            else:
                sy = None
            if ('end_year' in calc_climos):
                ey = calc_climos['end_year']
            else:
                ey = None
            cam_ts_data = cam_ts_data.sel(time=slice(sy,ey))
```

But I haven't had a chance to test. Also need to check whether this hits both the CAM case and the reference case.

**Comment by brianpm** [comment]
I'm working on a PR to fix this issue. One question for @nusbaume -- As part of this, I think I also need to address (maybe only partially) the issue in https://github.com/NCAR/ADF/issues/115.  In `averaging_example.py` I can check on whether `cam_overwrite_climo` is T/F on a case by case basis, but do we assume that will be provided as a list for multiple cases? If so, then I can just make a list. I'll assume this unless told otherwise.

**Comment by nusbaume** [comment]
@brianpm you are correct that we'll need to assume they are a list, but I also realized that it would be beneficial to not make the `cam_overwrite_climo` required, and instead have it default to `False` if not found.  I'll go ahead and make the relevant modifications and push them directly to PR #117 .  

As an FYI, I also might add the fixes for issue #120 as well, as they are pretty small and orthogonal to the rest of the code mods.  I'll then go ahead and merge this PR once I have tested everything to make sure it works.  Of course if you have any issues or concerns with this plan please let me know.  Thanks!

**Comment by brianpm** [comment]
That sounds good to me. Let me know if you need me to do anything along the way, otherwise I'll assume you are taking care of it. Thanks!!
